### ISSUE TITLE
"The diff algorithm was stopped early" when removing a large chunk (2k lines) of code

### ISSUE THREAD

**Issue by Mingpan** [microsoft/vscode#202242]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.86.0-insider (Universal)
- OS Version: MacOS Ventura

Steps to Reproduce:

1. Go to a large file, for example src/vs/editor/browser/widget/codeEditorWidget.ts
2. Remove all lines starting from the `CodeEditorWidget` class definition.
3. Add a comment `// class removed` where the class was removed.
4. Open diff with `advanced` diff algorithm set (which is also the default)
5. Notice the lag and the warning notification.

If one switches to `legacy` diff algorithm, the diff would render instantaneously, and no warnings are shown. 

I am not sure if this is expected and if it's because the algorithm is searching for the best match for the added comment within the original content. But it feels suboptimal and might create bigger issues within a merge editor for large files. 

Could you take a look? 

(CC @hediet )

Thanks!

**Comment by Mingpan** [comment]
<img width="1134" alt="diff-performance" src="https://github.com/microsoft/vscode/assets/15076498/1eaf58b1-0594-4b1d-885d-5bfcd13dc79e">

**Comment by jackreuter-vs** [comment]
I'm facing the same issue

**Comment by gayathridevi815** [comment]
Restarting the Vscode works for me!!!

**Comment by zeozeozeo** [comment]
Getting this issue with Roo Code also.

**Comment by vihanroyayy** [comment]
Getting the same issue now, when using github copilot (im on the pro plan 💀 )

<img width="716" height="277" alt="Image" src="https://github.com/user-attachments/assets/1a7edaac-3602-40ca-8204-439f4383a99f" />

**Comment by ram55git** [comment]
Any update on this issue, its recurring in github copilot in agent mode.

**Comment by lovemonkeyz** [comment]
> > Same problem here, any setting to increase that timeout?
> 
> I tried setting it really high, and to zero, in three different places, but to no avail.

Yep.. I noticed that as well. It didn't stop notifying until I restarted VSCode, but keeps coming back. I set the MS to 100000000, still stopped after 5000. Changed to legacy algo, no difference. Hopefully this can be fixed soon.
